shared/util/saga.js

```diff
@@ -98,22 +98,15 @@ function actionToAction<A, E>(pattern: RS.Pattern, f: (state: TypedState, action
 // It can also return all([put(action1), put(action2)]) to dispatch multiple actions
 function safeTakeEveryPure<A, R, FinalEffect, FinalErrorEffect>(
   pattern: RS.Pattern,
-  pureWorker: ((action: A, state: TypedState) => any) | ((action: A) => any),
+  pureWorker: (action: A, state: TypedState) => any,
```

Comment:
Would it work properly if you did `(action: A, state?: TypedState)`?

Reply:
I assume that would mean a different function signature for `pureWorker`, and all users of `safeTakeEveryPure` that needs the `state` would need to test if it's `undefined` or not.